src/lib.rs

```diff
@@ -51,11 +50,13 @@ impl Game {
         // Draw the outer circle.
         context
             .arc(75.0, 75.0, 50.0, 0.0, f64::consts::PI * 2.0)
-            .ok()?;
+            .map_err(|_| "Failed to draw an arc.")?;
 
         // Draw the mouth.
         context.move_to(110.0, 75.0);
-        context.arc(75.0, 75.0, 35.0, 0.0, f64::consts::PI).ok()?;
+        context
+            .arc(75.0, 75.0, 35.0, 0.0, f64::consts::PI)
+            .map_err(|_| "Failed to draw an arc.")?;
```

Comment:
Where does `map_err` come from?

Reply:
It's a method on Result used for transforming an error: https://doc.rust-lang.org/std/result/enum.Result.html#method.map_err